server/gamemodes/chess.py

```diff
@@ -218,7 +210,8 @@ def get_knight_moves(
                     file = knight.file + file_absolute * file_direction
                     victim = self.get_piece(rank, file)
                     if (not victim) or (victim.side != knight.side):
-                        yield rank, file
+                        if not self.hypothetical_check(knight, rank, file):
+                            yield rank, file
```

Comment:
```suggestion
                        yield rank, file
```
`hypothetical_check` is already called by `possible_moves`.